server/app/services/grid_scheduler.rb

```diff
@@ -66,18 +65,6 @@ def should_reschedule_service?(service)
 
   # @param [GridService] service
   # @return [Boolean]
-  def pending_deploys?(service)
-    service.grid_service_deploys.where(started_at: nil).count > 0
```

Comment:
With the changes to set `queued_at` instead of `started_at` (which might now remain unset if the service was stopped before the deploy started), this could have left the service in a state where it would never get rescheduled anymore.